aiida/orm/nodes/process/process.py

```diff
@@ -296,6 +296,19 @@ def exit_status(self):
         """
         return self.get_attribute(self.EXIT_STATUS_KEY, None)
 
+    def set_exit_code(self, exit_code):
+        """
+        Set the exit status and message via the exit code framework.
+
+        :param exit_code: a valid tuple set from an ExitCode instance
+        """
+
+        if isinstance(exit_code, tuple):
+            self.set_exit_status(exit_code[0])
+            self.set_exit_message(exit_code[1])
+        else:
+            raise ValueError('exit code is not a tuple set with ExitCode')
```

Comment:
this should be a `TypeError('exit code should be an instance of ExitCode')`
Actually, it is best to put this check at the top of the method using `aiida.common.lang.type_check`